Lib/subprocess.py

```diff
@@ -1307,15 +1307,15 @@ def _execute_child(self, args, executable, preexec_fn, close_fds,
                 try:
                     exception_name, hex_errno, err_msg = (
                             errpipe_data.split(b':', 2))
+                    err_msg = err_msg.decode(errors="surrogatepass")
                 except ValueError:
                     exception_name = b'SubprocessError'
                     hex_errno = b'0'
```

Comment:
Wouldn't that change the current behavior though, highly unlikely that someone is relying on the message looking like
```Bad exception data from child: bytearray(b'OSError:asdf')```

but this would change it to:
```Bad exception data from child: OSError: asdf```

If you think that's fine then I don't mind implementing it that way.